src/Shared/Buffers/ArrayPoolExtensions.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Buffers;
+
+internal static class ArrayPoolExtensions
+{
+    /// <summary>
+    /// Clears the specified range and returns the array to the pool.
+    /// </summary>
+    public static void Return<T>(this ArrayPool<T> pool, T[] array, int lengthToClear)
```

Comment:
What's the point of this overload? Shouldn't we just always use the `ReturnAndClearReferences` one?

Reply:
It's for cases were it's known that `T` is a reference type.

Do you think `ReturnAndClearReferences` should be used everywhere? I assume `RuntimeHelpers.IsReferenceOrContainsReferences<T>()` check is optimized away by JIT.